modules/python/src2/cv2_convert.cpp

```diff
@@ -139,11 +139,13 @@ bool pyopencv_to(PyObject* o, Mat& m, const ArgInfo& info)
                typenum == NPY_FLOAT ? CV_32F :
                typenum == NPY_DOUBLE ? CV_64F :
                typenum == NPY_BOOL ? CV_Bool :
+               typenum == NPY_INT64 ? CV_64S :
+               typenum == NPY_UINT64 ? CV_64U :
                -1;
 
     if( type < 0 )
     {
-        if( typenum == NPY_INT64 || typenum == NPY_UINT64 || typenum == NPY_LONG )
```

Comment:
numpy interprets int as int64 by default. The change generates Mats with CV_64U(S) in most of places instead of CV_32S. It breaks a lot of code. We tried similar changes in https://github.com/opencv/opencv/pull/26514 and decided to reject it for now.

Reply:
I reverted Python changes, added needed cases to C++ test (and took changes from #26852 to pass test)